plugins/redmine_bugs_ruby_lang/lib/redmine_mailing_list_integration/redmine_ext/mailer.rb

```diff
@@ -1,74 +1,51 @@
-module MailingListIntegrationMailer
+module MailingListIntegration
+  def deliver_issue_add(issue)
+    deliver_issue_add_to_mailing_lists(issue) unless issue.originates_from_mail?
+    super
+  end
 
-  def issue_add(user, issue)
-    mailing_lists = issue.project.mail_routes_for_issue(issue)
+  def deliver_issue_edit(journal)
+    deliver_issue_edit_to_mailing_lists(journal) unless journal.originates_from_mail?
+    super
+  end
 
-    record_message(issue, nil, mailing_lists)
+  def deliver_attachments_added(attachments)
+    deliver_attachments_added_to_mailing_lists(attachments) if attachments.first.container_type == "Issue"
+    super
+  end
 
-    m = super(user, issue)
+  private
 
-    m.header[:to] = mailing_lists.map(&:address)
-    m.header[:subject] = "[#{issue.project.name} #{issue.tracker.name}##{issue.id}] #{issue.subject}"
-    m
+  def deliver_issue_add_to_mailing_lists(issue)
+    mail = issue_add(issue.author, issue)
+    mailing_lists = issue.project.mail_routes_for_issue(issue)
+    deliver_to_mailing_lists(mail, mailing_lists, issue: issue)
   end
 
-  def issue_edit(user, journal)
+  def deliver_issue_edit_to_mailing_lists(journal)
     issue = journal.issue
+    mail = issue_edit(issue.author, journal)
     mailing_lists = issue.project.mail_routes_for_issue(issue)
-
-    record_message(issue, journal, mailing_lists)
-
-    m = super(user, journal)
-
-    m.header[:to] = mailing_lists.map(&:address)
-    m.header[:subject] = "[#{issue.project.name} #{issue.tracker.name}##{issue.id}] #{issue.subject}"
-    m
+    deliver_to_mailing_lists(mail, mailing_lists, issue: issue, journal: journal)
   end
 
-  def attachments_added(user, attachments)
-    mailing_lists = attatchments.first.container.project.mail_routes_for_attachments(attachments)
```

Comment:
btw there was a typo attatchments vs attachments

Reply:
Interesting.  I wonder if that means this code is rarely called, or if the delivery jobs have been failing but no one has noticed.